add a benchmark that compares against IMOD on the shrec dataset

McHaillet opened the pull request:
@dmichalak thanks for asking about the benchmark, writing it down in a notebook made me immediately spot a bug! If you go to the branch on my fork, you should be able to see a plot that shows the comparison with IMOD. Let me know if its clear like this :)

This PR adds:
* a notebook that quantitatively compares against the alignments provided by IMOD on the SHREC'21 dataset
* a bugfix for the indexing of the branches that I found through plotting this, in other words immediately got use out of adding it :laughing: 

Not sure in general what the best way is to provide benchmarks on our implementations. I guess notebooks like this where concise preprocessing steps are given from raw downloadable data are fairly straightforward though.

dmichalak commented:
I'm glad asking about it itself was helpful! 

Here are my comments on the notebook:

I think the notebook format with comments is a nice way to show benchmarking especially with showing what IMOD commands are used. It also sets up a quick framework for a user to do a comparison with some of their own data.

Seeing the IMOD parameters explicitly did make me look up if they are the default ones used in something like `batchruntomo` and `etomo`. For clarity, it could be helpful to have a comment saying they are the default and that `-radius2` is equivalent to `low_pass_cutoff` (if they actually are). 

In cell 3, I don't see where the intended multiplication of -1 is done for the forward-projection model. 
```python
tilt_axis_angle = 0.0
# invert to match reconstruction with SHREC grand_model
tilt_angles = list(reversed(tilt_angles))
# divide by two for 2x downsampling of tilt-series
# invert the shifts because we employ a forward projection model!
ground_truth_shifts = torch.tensor([y_shift, x_shift]).T / 2

# also multiply with -1 for forward-projection model
torch_xcorr_shifts = tiltxcorr(
    tilt_series,
    tilt_angles,
    tilt_axis_angle,
    low_pass_cutoff=0.25,
)
```

McHaillet commented:
Thanks for reviewing :)

> Seeing the IMOD parameters explicitly did make me look up if they are the default ones used in something like `batchruntomo` and `etomo`. For clarity, it could be helpful to have a comment saying they are the default and that `-radius2` is equivalent to `low_pass_cutoff` (if they actually are).

Good point! They are the default values, I added a comment for explanation. BTW, torch-tiltxcorr internally also does a high-pass filter and a smooth drop-off, but those parameters are not exposed. I added a comment for that as well.

> In cell 3, I don't see where the intended multiplication of -1 is done for the forward-projection model.

Thanks for spotting, that was a leftover from copying the code from a reconstruction workflow. If you apply the shifts here directly to the images you should get an aligned tilt-series. However, for a reconstruction algorithm the alignment parameters rather refer to the forward projection model, so there I had to invert them.
So, I removed those comments
